doc/version-history.rst

```diff
@@ -4,6 +4,82 @@
+
+  This enumeration will be removed once we release a version of ts-idl with it.
+  For now it will be kept here for backward compatibility.
+
+* In ``utils/exceptions``:
+
+  * Rename exception ``UnableToFindTarget`` to ``UnableToFindTargetError``.
+
+  * Add new exception ``FailedToQueueTargetsError``.
+
+* In ``scheduler_csc``:
+
+  * Background tasks management refactor:
+
+    Add a ``_tasks`` dictionary to store all background tasks that need to be managed by the CSC when going in and out of disabled state.
+    Move tasks ``target_production_task`` and ``telemetry_loop_task`` to this new dictionary.
+    Add methods ``_stop_all_background_tasks`` and ``_stop_background_task`` to handle stopping background tasks.
+
+    In ``begin_disable``, replace inline management of stopping ``target_production_task`` with call to ``_stop_all_background_tasks``.
+
+    In ``handle_summary_state``, replace management of ``telemetry_loop_task`` with call to ``_stop_all_background_tasks``.
+
+  * Add detailed state transition facility.
+
+    The detailed state transition is mostly managed using a decorator.
+    Methods decorated with ``set_detailed_state`` will first acquire a detailed state lock, guaranteeing no other detailed state transition can happen at the same time.
+    Then they assert that the detailed state is ``RUNNING``, otherwise they cannot go into one of the operational sub states.
+    Next, the state switches to the desired new detailed state, executes the decorated coroutine and, once it is done, returns the system to ``RUNNING``.
+    There are also methods to send the CSC from IDLE to ``RUNNING`` and vise-versa.
+
+  * Refactoring ``advance_target_loop``.
+
+    * Extract method to add targets to the queue into a new method ``queue_targets``.
+      This method will switch the detailed state to ``QUEUEING_TARGET`` and execute the operations previously done inline in ``advance_target_loop``.
+      If the method fails to producer targets and fails to find a target in the future, it raises a new exception ``FailedToQueueTargetsError``.
+
+    * Change how ``generate_target_queue`` and ``compute_predicted_schedule`` runs.
+
+      Instead of calling them sequentially, which causes a delay in producing targets, call ``generate_target_queue`` first and then, while waiting for targets to execute, call ``compute_predicted_schedule``.
+      This new way improves over the previous implementation but still leave room for conditions where ``compute_predicted_schedule`` can take longer to execute than the targets to be observed and then lead to additional delay.
+      Nevertheless, conditions like this were only observed with the AuxTel configuration, which is a lot heavier than the one for the main telescope, and because there was a bug in the predictive loop which would continue to compute targets until the queue was filled up.
+
+    * Update to python>3.11 compatibility, stop using coroutine in ``asyncio.wait``.
```

Comment:
Python 3.11? I thought we were still on 3.10.

Reply:
yes.. it is ia future proofing feature.. Using coroutine directly in `asyncio.wait` will be deprecated in 3.11 so I'm fixing it now.

Reply:
Good to know. Thanks!

Reply:
I'll explain it better.